File: app/static/js/permissions.js
class Permissions {
  static TABLE = {
    FILTER_SORT_SEARCH: ["manager", "admin"],
    EDIT_PRODUCTS: ["admin"],
    VIEW_ORDERS: ["manager", "admin"],
    EDIT_ORDERS: ["admin"],
  };

  static has(role, permissionKey) {
    const allowed = Permissions.TABLE[permissionKey];
    if (!role || !allowed) return false;
    return allowed.includes(role);
  }
}

window.Permissions = Permissions;

File: app/core/permissions.py
"""Единая таблица прав по ролям — зеркало app/static/js/permissions.js.
Добавляйте новые права сюда и там одновременно."""

PERMISSIONS = {
    "FILTER_SORT_SEARCH": ("manager", "admin"),
    "EDIT_PRODUCTS": ("admin",),
    "VIEW_ORDERS": ("manager", "admin"),
    "EDIT_ORDERS": ("admin",),
}


def has_permission(role: str | None, permission_key: str) -> bool:
    if not role or permission_key not in PERMISSIONS:
        return False
    return role in PERMISSIONS[permission_key]
